### PR TITLE
feat: multiline function definition syntax

### PR DESCRIPTION
**Pull request by aljazerzen**

### CODE REVIEW

**Comment by max-sixty** [comment]
I would have preferred #819 with a preceding `\` over something specific to functions. Do you think if we do that, then we can remove this? Or you like this per se?

**Comment by aljazerzen** [comment]
This would be hard to do in general, as new lines are parsed slightly differently in different cases.

Maybe it's doable, but I wanted to add back the `func` for the function definition since there are some cases where it is hard to parse an expression if you are not sure if it is a function definition or not. Unfortunately, I don't remember the example.
